scripts/dts/python-devicetree/src/devicetree/edtlib.py

```diff
@@ -1131,6 +1132,64 @@ def _check_undeclared_props(self):
                      f"{self.edt.dts_path}, but is not declared in "
                      f"'properties:' in {self.binding_path}")
 
+    def _init_ranges(self):
+        # Initializes self.ranges
+        node = self._node
+
+        self.ranges = []
+
+        if "ranges" not in node.props:
+            return
+
+        child_address_cells = node.props.get("#address-cells")
+        parent_address_cells = _address_cells(node)
+        if not child_address_cells:
+            child_address_cells = 2 # Default value per DT spec.
+        else:
+            child_address_cells = child_address_cells.to_num()
+        child_size_cells = node.props.get("#size-cells")
+        if not child_size_cells:
+            child_size_cells = 1 # Default value per DT spec.
+        else:
+            child_size_cells = child_size_cells.to_num()
+
+        # Number of cells for one translation 3-tuple in 'ranges'
+        entry_cells = child_address_cells + parent_address_cells + child_size_cells
+
+        for raw_range in _slice(node, "ranges", 4*entry_cells,
+                "4*(<#address-cells> (= {}) + "
+                "<#address-cells for parent> (= {}) + "
+                "<#size-cells> (= {}))"
+                .format(child_address_cells, parent_address_cells,
+                    child_size_cells)):
+
+            range_ = Range()
+            range_.node = self
+            if child_address_cells == 0:
+                range_.child_bus_flags = None
+                range_.child_bus_addr = None
+            elif child_address_cells == 3:
+                # When used as PCIe ranges, the first entry can be some flags
+                range_.child_bus_flags = to_num(raw_range[:4])
+                range_.child_bus_addr = to_num(raw_range[4:4*(child_address_cells)])
+            else:
+                range_.child_bus_flags = None
+                range_.child_bus_addr = to_num(raw_range[:4*child_address_cells])
+            if parent_address_cells == 0:
+                range_.parent_bus_addr = None
+            else:
+                range_.parent_bus_addr = to_num(raw_range[(4*child_address_cells):\
+                                            (4*child_address_cells + 4*parent_address_cells)])
+            if child_size_cells == 0:
+                range_.length = None
+            else:
+                range_.length = to_num(raw_range[(4*child_address_cells + \
+                                                    4*parent_address_cells):])
+
+            self.ranges.append(range_)
+
+        _add_names(node, "ranges", self.ranges)
```

Comment:
I'm not familiar with names being used for the ranges property; can you please provide a reference to explain what's going on here?

Reply:
Isn't `add_names` necessary to be used from `gen_defines` ?